### PR TITLE
Clean up class Ions, add test for it

### PR DESCRIPTION
**Pull request by jeanlucf22**

### CODE REVIEW

**Comment by jeanlucf22** [comment]
@dreamer2368 This should help you with PR #274

File: src/KBprojectorSparse.cc [hunk 1]
```diff
@@ -103,7 +103,7 @@ void KBprojectorSparse::setNLindex(
 
     for (int i = 0; i < size_nl; i++)
     {
-        assert(i < lnumpt);
+        // assert(i < lnumpt);
```
Comment (oseikuffuor1) [inline src/KBprojectorSparse.cc:106]: Should we just clean this out, so there's no question whether it should be there or not?

File: src/KBprojectorSparse.cc [hunk 2]
```diff
@@ -960,7 +960,7 @@ bool KBprojectorSparse::setIndexesAndProjectors()
             // get "pvec" and "is_in_domain"
             int icount = get_index_array(pvec, iloc, index_low, index_high);
             assert(icount <= nl3);
-            assert(icount <= mymesh->npointsPatch());
+            //            assert(icount <= mymesh->npointsPatch());
```
Comment (oseikuffuor1) [inline src/KBprojectorSparse.cc:963]: Same comment as above

**Review by oseikuffuor1 — APPROVED** [review]
Hi Jean-Luc, This looks very good. Thanks! I just left a couple minor comments about clearing the commented-out asserts

File: src/KBprojectorSparse.cc (hunk 1 above)
Comment (jeanlucf22) [reply src/KBprojectorSparse.cc:106]: I actually still have questions about those asserts, that's why I decided to keep them...

**Pull request merged**